src/plugins/pcr_primer_design_for_dna_assembly/src/PCRPrimerProductTable.cpp

```diff
@@ -0,0 +1,92 @@
+/**
+ * UGENE - Integrated Bioinformatics Tools.
+ * Copyright (C) 2008-2021 UniPro <ugene@unipro.ru>
+ * http://ugene.net
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+#include <U2Core/AnnotationSelection.h>
+#include <U2Core/AnnotationTableObject.h>
+#include <U2Core/DNASequenceSelection.h>
+#include <U2Core/U2SafePoints.h>
+
+#include <U2View/ADVSequenceObjectContext.h>
+#include <U2View/AnnotatedDNAView.h>
+
+#include "PCRPrimerProductTable.h"
+#include "src/tasks/PCRPrimerDesignForDNAAssemblyTask.h"
+
+namespace U2 {
+
+ PCRPrimerProductTable::PCRPrimerProductTable(QWidget *parent)
+    : QTableWidget(parent) {
+     setColumnCount(2);
+     setRowCount(8);
```

Comment:
As it's said in the issue, "It's possible, that some of fragments won't exist. ", so it's not correct to be connected to the number of rows!